src/sagemaker_container_support/environment.py

```diff
@@ -0,0 +1,607 @@
+                                For example, 'algo-1'.
+                            -  hosts: The list of names of all containers on the container network,
+                                sorted lexicographically. For example, `['algo-1', 'algo-2', 'algo-3']`
+                                for a three-node cluster.
+    """
+    return read_json(os.path.join(INPUT_CONFIG_PATH, RESOURCE_CONFIG_FILE))
+
+
+def read_input_data_config():  # type: () -> Dict[str, Dict[str, str]]
+    """Read the input data configuration from /opt/ml/input/config/inputdataconfig.json.
+
+        For more information about inpudataconfig.json:
+https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-training-algo.html#your-algorithms-training-algo-running-container-dist-training
+
+    Returns:
+            input_data_config (dict[string, object]): contents from /opt/ml/input/config/inputdataconfig.json.
+
+                                For example, suppose that you specify three data channels (train, evaluation, and
+                                validation) in your request. This dictionary will contain:
+
+                                {'train': {
+                                    'ContentType':  'trainingContentType',
+                                    'TrainingInputMode': 'File',
+                                    'S3DistributionType': 'FullyReplicated',
+                                    'RecordWrapperType': 'None'
+                                },
+                                'evaluation' : {
+                                    'ContentType': 'evalContentType',
+                                    'TrainingInputMode': 'File',
+                                    'S3DistributionType': 'FullyReplicated',
+                                    'RecordWrapperType': 'None'
+                                },
+                                'validation': {
+                                    'TrainingInputMode': 'File',
+                                    'S3DistributionType': 'FullyReplicated',
+                                    'RecordWrapperType': 'None'
+                                }}
+
+                                You can find more information about /opt/ml/input/config/inputdataconfig.json here:
+                                https://docs.aws.amazon.com/sagemaker/latest/dg/your-algorithms-training-algo.html#your-algorithms-training-algo-running-container-inputdataconfig
+    """
+    return read_json(os.path.join(INPUT_CONFIG_PATH, INPUT_DATA_CONFIG_FILE))
+
+
+def channel_path(channel):  # type: (str) -> str
+    """ Returns the directory containing the channel data file(s) which is:
+    - <self.base_dir>/input/data/<channel>
+
+    For more information about channels: https://docs.aws.amazon.com/sagemaker/latest/dg/API_Channel.html
+
+    Returns:
+        (str) The input data directory for the specified channel.
+    """
+    return os.path.join(INPUT_DATA_PATH, channel)
+
+
+def gpu_count():  # type: () -> int
```

Comment:
gpu_count -> num_gpus? Same as in train's function signature.

Reply:
Because of design decisions. @winstonaws and I came with the following design:

```Environment``` will be used by our customers. It is just a class with getters and setters well documented explaining how sagemaker environment works.

The entire framework is made by composition of small functions. These small functions allows users adhere to high level solution or just use the small functions to extend their needs.

The entire framework was refactored from the bottom up to that objective.